ext/scheduler/airflow/bucket/ossblob/ossblob.go

```diff
@@ -0,0 +1,405 @@
+	}
+
+	if opts.ContentEncoding != "" {
+		req.ContentEncoding = &opts.ContentEncoding
+	}
+
+	if opts.CacheControl != "" {
+		req.CacheControl = &opts.CacheControl
+	}
+
+	if len(opts.Metadata) > 0 {
+		req.Metadata = opts.Metadata
+	}
+
+	w := NewOSSWriter(req, pr, pw)
+
+	go func() {
+		defer close(w.doneCh)
+
+		_, err := b.client.PutObject(ctx, &w.req)
+		w.err = err
+		w.pr.Close()
+	}()
+
+	return w, nil
+}
+
+// Delete implements driver.Delete.
+func (b *ossBucket) Delete(ctx context.Context, key string) error {
+	_, err := b.client.DeleteObject(ctx, &oss.DeleteObjectRequest{
+		Bucket: &b.bucket,
+		Key:    &key,
+	})
+
+	return err
+}
+
+// Attributes implements driver.Attributes.
+func (b *ossBucket) Attributes(ctx context.Context, key string) (*driver.Attributes, error) {
+	objMeta, err := b.client.HeadObject(ctx, &oss.HeadObjectRequest{
+		Bucket: &b.bucket,
+		Key:    &key,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	attrs := driver.Attributes{
+		CacheControl:       safeGet(objMeta.CacheControl),
+		ContentDisposition: safeGet(objMeta.ContentDisposition),
+		ContentEncoding:    safeGet(objMeta.ContentEncoding),
+		Metadata:           objMeta.Metadata,
+		Size:               objMeta.ContentLength,
+		ModTime:            safeGet(objMeta.LastModified),
+		MD5:                []byte(safeGet(objMeta.ContentMD5)),
+		ETag:               safeGet(objMeta.ETag),
+		AsFunc: func(i interface{}) bool {
+			p, ok := i.(**oss.HeadObjectResult)
+			if !ok {
+				return false
+			}
+			*p = objMeta
+			return true
+		},
+	}
+
+	return &attrs, nil
+}
+
+// Close implements driver.Close.
+func (ossBucket) Close() error {
+	return nil
+}
+
+// SignedURL implements driver.SignedURL
+func (b *ossBucket) SignedURL(ctx context.Context, key string, opts *driver.SignedURLOptions) (string, error) {
+	res, err := b.client.Presign(ctx, &oss.GetObjectRequest{
+		Bucket: &b.bucket,
+		Key:    &key,
+	}, oss.PresignExpires(opts.Expiry))
+	if err != nil {
+		return "", err
+	}
+
+	return res.URL, nil
+}
+
+// Copy implements driver.Copy
+func (b *ossBucket) Copy(ctx context.Context, dstKey, srcKey string, _ *driver.CopyOptions) error {
+	copier := oss.NewCopier(b.client)
+	_, err := copier.Copy(ctx, &oss.CopyObjectRequest{
+		Bucket:       &b.bucket,
+		SourceBucket: &b.bucket,
+		Key:          &dstKey,
+		SourceKey:    &srcKey,
+	})
+
+	return err
+}
+
+func openBucket(_ context.Context, cfg *oss.Config, bucketName string) (*ossBucket, error) {
+	if cfg == nil {
+		return nil, errors.New("ossblob.openBucket: oss config are required")
+	}
+	if cfg.CredentialsProvider == nil {
+		return nil, errors.New("ossblob.openBucket: credentials provider is required")
+	}
+	if bucketName == "" {
+		return nil, errors.New("ossblob.openBucket: bucketName is required")
+	}
+
+	client := oss.NewClient(cfg)
+
+	return NewOSSBucket(client, bucketName), nil
+}
+
+func OpenBucket(ctx context.Context, cfg *oss.Config, bucketName string) (*blob.Bucket, error) {
```

Comment:
is there any gocloud implementation for `ossblob`? instead of creating new one, it's better to use from official sdk one